### DISCUSSION TITLE
JWT Authentication with Middleware (Edge Functions) Next-Auth

### DISCUSSION THREAD

**Discussion by juanzgc** [vercel/next.js#31093] [Help]
Considering that now it is possible to authenticate users with a Middleware using Edge Functions, how can we pass that session data as a prop?

![image](https://user-images.githubusercontent.com/24414523/140619982-28259b24-e8b3-42cb-bdf8-d95504bc833e.png)

The image above is from @balazsorban44 (Thank you btw!)

However, in the client side I don't want to make another request to get the Authenticated Session, instead it should be passed as a prop. Any idea how to pass middleware data as a prop to pages.

-- /pages
-- / pages/_middleware       <-- Handle Authentication
-- -- -- /pages/index
-- -- -- /pages/page1
-- -- -- /pages/page2

In this structure, the idea is that all pages should require authentication, and so I would want to pass the auth prop to the three pages: `index`, `page1`, & `page2`

**Comment by ghost** [comment]
I don't think middleware has support for that now: https://github.com/vercel/next.js/discussions/29750#discussioncomment-1537361
